@here/harp-materials/lib/SolidLineMaterial.ts

```diff
@@ -589,38 +588,16 @@ export class SolidLineMaterial extends RawShaderMaterial
             }
             if (params.fog !== undefined) {
                 this.fog = params.fog;
+                this.invalidateFog();
             }
             this.offset = params.offset ?? 0;
 
             // ShaderMaterial overrides requires invalidation cause super c-tor may set this
             // properties before related `defines` and `uniforms` were created.
-            this.invalidateFog();
             this.invalidateOpacity();
         }
     }
 
-    /**
-     * Overrides THREE.Material.fog flag to add support for custom shader.
-     *
-     * @param enable - Whether we want to enable the fog.
-     */
-    set fog(enable: boolean) {
-        this.m_fog = enable;
-        // Function may be called from THREE.js cause we override setter,
-        // in this case defines are not yet initialized and require late invalidation in
-        // SolidLineMaterial c-tor.
-        if (this.defines !== undefined) {
-            setShaderMaterialDefine(this, "USE_FOG", enable);
-        }
-    }
```

Comment:
The problem now is that the material define is not updated when fog is enabled/disabled, so the shader won't change even if it's recompiled.
Since the fog settings are controlled only from MapViewFog, we can add a new method setFog in RawShaderMaterial that sets this.fog and then calls invalidateFog().
Then we just need to change MapViewFog.setFogInRawShaderMaterials() so that it calls setFog in raw materials.